src/lsptoolshost/miscellaneousFileNotifier.ts

```diff
@@ -13,11 +13,35 @@ import { languageServerOptions } from '../shared/options';
 const SuppressMiscellaneousFilesToastsOption = 'dotnet.server.suppressMiscellaneousFilesToasts';
 const NotifiedDocuments = new Set<string>();
 
+const NotificationDelay = 2 * 1000;
+
+let _notifyTimeout: NodeJS.Timeout | undefined;
+let _documentUriToNotify: vscode.Uri | undefined;
+
 export function registerMiscellaneousFileNotifier(
     context: vscode.ExtensionContext,
     languageServer: RoslynLanguageServer
 ) {
     languageServer._projectContextService.onActiveFileContextChanged((e) => {
+        // Whether we have refreshed the active document's project context.
+        let isNotifyPass = false;
+
+        if (_notifyTimeout) {
+            // If we have changed active document then do not notify about the previous one.
+            clearTimeout(_notifyTimeout);
+            _notifyTimeout = undefined;
+        }
+
+        if (_documentUriToNotify) {
+            if (e.uri.toString() === _documentUriToNotify.toString()) {
+                // We have rerequested project contexts for the active document
+                // and we can now notify if the document isn't part of the workspace.
+                isNotifyPass = true;
```

Comment:
do we need to add to NotifiedDocuments? Looks like we're no longer adding to it anywhere

Reply:
nvm I see the update